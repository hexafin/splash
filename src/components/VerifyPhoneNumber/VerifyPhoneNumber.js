import React, {Component} from "react"
import {
    View,
    Text,
    StyleSheet,
    Image,
    TouchableOpacity,
    Keyboard,
    Alert,
    KeyboardAvoidingView,
} from "react-native"
import {colors} from "../../lib/colors"
import {defaults, icons} from "../../lib/styles"
import FlatBackButton from "../universal/FlatBackButton"
import Button from "../universal/Button"
import NumericCodeInput from "../universal/NumericCodeInput"

class VerifyPhoneNumber extends Component {

    constructor(props) {
        super(props)
        this.state = {
            code: ""
        }
    }

    componentDidUpdate(prevProps) {
      if((this.props.confirmError != prevProps.confirmError && this.props.confirmError) || (this.props.claimError != prevProps.claimError && this.props.claimError)) {
        Alert.alert('An error occurred!', 'Sorry about this. Our team has been notified and we should fix this shortly!', [
            {
                text: 'Cancel',
                onPress: () => console.log('Cancel Pressed'),
                style: 'cancel'
            }, {
                text: 'Ok',
                onPress: () => console.log('OK Pressed')
            }
        ])
      }
    }

    render() {

        return (<KeyboardAvoidingView style={styles.container} behavior={"padding"}>

            <View style={styles.body}>

                <Text style={styles.title}>
                    Enter the 6-digit code we{"\n"}
                    just texted you
                </Text>

                <View>
                    <NumericCodeInput size={6} autoFocus={true} callback={(code) => {
                        this.setState((prevState) => {
                            return {
                                ...prevState,
                                code: code
                            }
                        })
                    }}/>

                    <View style={styles.resendWrapper}>
                        <Text style={styles.resendText}>Didn{"'"}t get your code?</Text>
                        <Button
                            onPress={() => {
                                this.props.SmsAuthenticate(this.props.phoneNumber, this.props.countryName)
                            }} style={styles.resendButton} title={"Resend"}
                            primary={false} loading={this.props.isSmsAuthenticating} small/>
                    </View>
                </View>

                <Button onPress={() => {
                        // TODO: sms authentication function
                        Keyboard.dismiss()
                        this.props.SmsConfirm(this.state.code)
                    }} style={styles.footerButton} title={"Claim splashtag"}
                    primary={true}
                    loading={this.props.isSmsConfirming}
                    disabled={this.state.code.length < 6}/>

            </View>

            <FlatBackButton onPress={() => {
                Keyboard.dismiss()
                this.props.navigation.navigate("EnterPhoneNumber")
            }}/>
        </KeyboardAvoidingView>)

    }

}

const styles = StyleSheet.create({
    container: {
        ...defaults.container,
        justifyContent: "space-between",
        position: "relative",
        paddingBottom: 27,
        paddingTop: 80
    },
    body: {
        flex: 1,
        padding: 20,
        flexDirection: "column",
        justifyContent: "space-between"
    },
    title: {
        fontSize: 23,
        color: colors.nearBlack,
        marginTop: 20,
        marginBottom: 20,
        textAlign: "center"
    },
    resendWrapper: {
        flexDirection: 'row',
        marginTop: 20,
        justifyContent: 'center',
        alignItems: 'center',
    },
    resendText: {
        fontSize: 16,
        color: colors.lightGray
    },
    resendButton: {
        width: 90,
        marginLeft: 10,
    }
})

export default VerifyPhoneNumber
