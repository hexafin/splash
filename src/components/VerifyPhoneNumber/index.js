import VerifyPhoneNumber from "./VerifyPhoneNumber";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { SmsConfirm, SmsAuthenticate } from "../../redux/waitlist/actions";

const mapStateToProps = state => {
	return {
		waitlisted: state.user.waitlisted,
		username: state.waitlist.username,
		phoneNumber: state.waitlist.phoneNumber,
		countryName: state.waitlist.countryName,
		confirmError: state.waitlist.errorSmsConfirming,
		claimError: state.waitlist.errorClaimingUsername,
		isSmsConfirming: state.waitlist.isSmsConfirming
	};
};

const mapDispatchToProps = dispatch => {
	return bindActionCreators(
		{
			SmsConfirm,
			SmsAuthenticate
		},
		dispatch
	);
};

export default connect(mapStateToProps, mapDispatchToProps)(VerifyPhoneNumber);
