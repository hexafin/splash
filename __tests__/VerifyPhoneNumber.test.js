import "react-native";
import React from "react";
import VerifyPhoneNumber from "../src/components/VerifyPhoneNumber/VerifyPhoneNumber";

it("Verify Phone Number renders correctly", () => {
	const snap = shallow(
		<VerifyPhoneNumber
			navigation={{
				state: {
					params: {}
				},
				setParams: () => {},
				navigate: () => {}
			}}
		/>
	);
	expect(snap).toMatchSnapshot();
});
