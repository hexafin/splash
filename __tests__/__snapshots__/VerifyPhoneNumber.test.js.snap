// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify Phone Number renders correctly 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "space-between",
      "paddingBottom": 27,
      "paddingTop": 80,
      "position": "relative",
    }
  }
>
  <Component
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "padding": 20,
      }
    }
  >
    <Component
      style={
        Object {
          "color": "#333333",
          "fontSize": 23,
          "marginBottom": 20,
          "marginTop": 20,
          "textAlign": "center",
        }
      }
    >
      Enter the 6-digit code we
      

      just texted you
    </Component>
    <Component>
      <NumericCodeInput
        autoFocus={true}
        callback={[Function]}
        size={6}
      />
      <Component
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 20,
          }
        }
      >
        <Component
          style={
            Object {
              "color": "#ADADAD",
              "fontSize": 16,
            }
          }
        >
          Didn
          '
          t get your code?
        </Component>
        <Button
          onPress={[Function]}
          primary={false}
          small={true}
          style={
            Object {
              "marginLeft": 10,
              "width": 90,
            }
          }
          title="Resend"
        />
      </Component>
    </Component>
    <Button
      disabled={true}
      onPress={[Function]}
      primary={true}
      title="Finish signup"
    />
  </Component>
  <FlatBackButton
    onPress={[Function]}
  />
</KeyboardAvoidingView>
`;
